GMX-Bots/Large-Open-and-Increase-Positions/src/agent.spec.ts

```diff
@@ -0,0 +1,542 @@
+        updatePositionEventLog.data,
+        ...updatePositionEventLog.topics
+      );
+
+    findings = await handleTransaction(txEvent);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("should detect finding for IncreasePosition event whose size delta is large", async () => {
+    const increasePositionEventLog = MOCK_IFACE.encodeEventLog(increasePositionEventFragment, [
+      updatePositionKey,
+      MOCK_ACCOUNT,
+      MOCK_TOKEN,
+      MOCK_TOKEN,
+      0,
+      largeSizeDelta,
+      true,
+      0,
+      0,
+    ]);
+
+    const updatePositionEventLog = MOCK_IFACE.encodeEventLog(updatePositionEventFragment, [
+      increasePositionKey,
+      largePositionSize2,
+      0,
+      0,
+      0,
+      0,
+      0,
+    ]);
+
+    txEvent = new TestTransactionEvent()
+      .addAnonymousEventLog(
+        mockNetworkManager.vaultAddress,
+        increasePositionEventLog.data,
+        ...increasePositionEventLog.topics
+      )
+      .addAnonymousEventLog(
+        mockNetworkManager.vaultAddress,
+        updatePositionEventLog.data,
+        ...updatePositionEventLog.topics
+      );
+
+    findings = await handleTransaction(txEvent);
+    expect(findings).toStrictEqual([
+      mockCreateFinding(
+        MOCK_ACCOUNT,
+        mockNetworkManager.vaultAddress,
+        updatePositionKey,
+        updatePositionKey,
+        largePositionSize2,
+        largeSizeDelta,
+        ethers.BigNumber.from(largePositionSize2).sub(ethers.BigNumber.from(largeSizeDelta))
+      ),
+    ]);
+  });
+
+  it("should detect finding for UpdatePosition event whose position size is large", async () => {
+    const increasePositionEventLog = MOCK_IFACE.encodeEventLog(increasePositionEventFragment, [
+      updatePositionKey,
+      MOCK_ACCOUNT,
+      MOCK_TOKEN,
+      MOCK_TOKEN,
+      0,
+      largeSizeDelta,
+      true,
+      0,
+      0,
+    ]);
+
+    const updatePositionEventLog = MOCK_IFACE.encodeEventLog(updatePositionEventFragment, [
+      updatePositionKey,
+      largePositionSize,
+      0,
+      0,
+      0,
+      0,
+      0,
+    ]);
+
+    txEvent = new TestTransactionEvent()
+      .addAnonymousEventLog(
+        mockNetworkManager.vaultAddress,
+        updatePositionEventLog.data,
+        ...updatePositionEventLog.topics
+      )
+      .addAnonymousEventLog(
+        mockNetworkManager.vaultAddress,
+        increasePositionEventLog.data,
+        ...increasePositionEventLog.topics
+      );
+    findings = await handleTransaction(txEvent);
+    expect(findings).toStrictEqual([
+      mockCreateFinding(
+        MOCK_ACCOUNT,
+        mockNetworkManager.vaultAddress,
+        updatePositionKey,
+        updatePositionKey,
+        largePositionSize,
+        largeSizeDelta,
+        ethers.BigNumber.from(largePositionSize).sub(ethers.BigNumber.from(largeSizeDelta))
+      ),
+    ]);
+  });
+
+  it("should detect multiple findings for IncreasePosition event whose size delta is large", async () => {
```

Comment:
Instead of having this tests and the one one below merge them into one to check that multiple events of different types can be detected

Reply:
Noted, merged already